test.js

```diff
@@ -43,4 +55,14 @@ describe('Objects should do some shit', () => {
     console.log("\n\nproof.object\n", proof.object)
   })
+  it('Transaction Envelop V1', async () => {
+    let transaction = Transaction.fromHex('0x01f8a902850df847580082cd1b947c5a0ce9267ed19b22f8cae653f198e3e8daf09880b844a9059cbb00000000000000000000000083335e0c01afac5e02ff201ba0f5979ebc4aa93f00000000000000000000000000000000000000000000000340aad21b3b70000025a087e52ad60613b36f6d6dda62ff7aef013fc91340f6511c834a28f742fcfebf34a05e1bd17543f7e875e8872a3a19386e6428b661329e1ebdd38ad3fd14c256b196')
+    console.log("\n\ntransaction\n", transaction)
+    console.log("\n\ntransaction.object\n", transaction.object)
+  })
+  it('Transaction Envelop V2', async () => {
+    let transaction = Transaction.fromHex('0x02f8a902850df847580082cd1b947c5a0ce9267ed19b22f8cae653f198e3e8daf09880b844a9059cbb00000000000000000000000083335e0c01afac5e02ff201ba0f5979ebc4aa93f00000000000000000000000000000000000000000000000340aad21b3b70000025a087e52ad60613b36f6d6dda62ff7aef013fc91340f6511c834a28f742fcfebf34a05e1bd17543f7e875e8872a3a19386e6428b661329e1ebdd38ad3fd14c256b196')
+    console.log("\n\ntransaction\n", transaction)
+    console.log("\n\ntransaction.object\n", transaction.object)
+  })
```

Comment:
I dont believe the data is being encoded properly when building a new object because your test is outputting this:
```
transaction.object
 {
  chainId: '0x02',
  nonce: '0x0df8475800',
  maxPriorityFeePerGas: '0xcd1b',
  maxFeePerGas: '0x7c5a0ce9267ed19b22f8cae653f198e3e8daf098',
  gasLimit: '0x',
  to: '0xa9059cbb00000000000000000000000083335e0c01afac5e02ff201ba0f5979ebc4aa93f00000000000000000000000000000000000000000000000340aad21b3b700000',
  value: '0x25',
  data: '0x87e52ad60613b36f6d6dda62ff7aef013fc91340f6511c834a28f742fcfebf34',
  accessList: '0x5e1bd17543f7e875e8872a3a19386e6428b661329e1ebdd38ad3fd14c256b196',
  v: '0x',
  r: '0x',
  s: '0x'
}
```

Look at the 'to' address. Its not the right number of digits. Some of the other data here also looks wrong

Reply:
thats odd I will look into this looks like all the fields are messed up